### DISCUSSION TITLE
SMA and deep learning

### DISCUSSION THREAD

**Discussion by brunop26** [gama-platform/gama.old#3944] [Q&A] [answered]
Is it possible to integrate deep learning at agent or agent group?
Thank you

**Comment by benoitgaudou** [comment, marked as answer]
Hi,

I would be interesting about which use cases you need about that point.

What worked préviously (and hopefully should work again in March) was :
- train a model from Python
- use the trained model in GAMA to describe a part of the agents' behaviour.
E.g. in pedestrian simulation, we learnt the trajectory from data in
Python, and integrate the ML model in the GAMA simulation of pedestrians.

Another use case we are thinking about (and we have everything to do it I
think) is to learn the behavior of a complete GAMA model in Python.
The model could then be used as a Surrogate model of the agent-based model
in calibration tasks for example.
We need to write the Python pipeline to do that, in order to examplify this
use case.

But we could be interesting by other use cases....

Cheers

Benoit

**Reply by brunop26** [reply]
Thank you Benoit,
It's all about learning. Each agent has a role and thanks to ML it evolves over time. Let's imagine we want to build a house. We submit the plans to the architect agent (who is already pre-trained), but who also learns about the client's sensibilities. He then supervises the various craftsmen, who are themselves (pre-trained) but also learn from the work in progress. 
Best regards,
Bruno.

**Comment by brunop26** [comment]
Hello Benoit,
Do you think that some agents can be  pre-trained ? This oblige to train te model from Python .
Best regards,
